### PR TITLE
fix metadata_startup_script

### PR DESCRIPTION
**Pull request by edwardmedia**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/12776

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
compute: fixed the error when `metadata` and `machine_type` are updated while `metadata_startup_script` was already provided on `google_compute_instance`
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb [hunk 1]
```diff
@@ -6321,3 +6350,35 @@ resource "google_compute_instance" "foobar" {
 `, instance)
 }
 
+func testAccComputeInstance_metadataStartupScript(instance, machineType, metadata string) string {
```
Comment (roaks3) [inline mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:6353]: I saw some discussion on this in the GitHub issue, but do we actually need to change `machineType` to trigger the problem? If it is unrelated, we may want to remove it from the test.

**Review by roaks3 — COMMENTED** [review]
This looks great! I had one question on the tests, and also I think the CHANGELOG entry may need to be `compute: ...`, not `computer: ...`.

File: mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb (hunk 1 above)
Comment (edwardmedia) [reply mmv1/third_party/terraform/tests/resource_compute_instance_test.go.erb:6353]: Yes, that forces the vm to stop & restart. The issue happens only when the instance restart

**Comment by edwardmedia** [comment]
CHANGELOG updated as you pointed out. Thanks

**Review by roaks3 — APPROVED** [review]
LGTM, adding Stephen for follow-up review

**Review by melinath — COMMENTED** [review]
@edwardmedia it looks like the test is not exercising the changes to metadata.go.erb - that is, metadata.startup-script isn't being set in the test. Would it be possible to add a new test that does a series of updates that transition from using one to using the other?

**Pull request merged**